conmon-rs/server/src/rpc.rs

```diff
@@ -69,6 +69,34 @@ impl conmon::Server for Server {
         results.get().init_response().set_container_pid(pid as u32);
         Promise::ok(())
     }
+
+    fn exec_sync_container(
+        &mut self,
+        params: conmon::ExecSyncContainerParams,
+        mut results: conmon::ExecSyncContainerResults,
+    ) -> Promise<(), capnp::Error> {
+        let req = pry!(pry!(params.get()).get_request());
+        let id = pry!(req.get_id());
+        let _timeout = req.get_timeout();
+        debug!("Got exec sync container request for id {}", id);
+        if let Some(child) = self.children.get(id) {
+            debug!("found child with id {}", child.id);
+            let child_reaper = Arc::clone(self.reaper());
+            let command = self.generate_exec_sync_args(&params).unwrap();
+            let (exit_status, stdout, stderr) =
+                pry_err!(child_reaper.exec_sync(self.config().runtime(), command));
+            let mut resp = results.get().init_response();
+            resp.set_exit_code(exit_status.code().unwrap());
+            resp.set_stdout(stdout.as_str());
+            resp.set_stderr(stderr.as_str());
+        } else {
+            // container does not exist
```

Comment:
in this situation we should error. I would reverse the blocks and return with an error if None here